### ISSUE TITLE
Please explain why "all operators should go in the newline as first character"

### ISSUE THREAD

**Issue by dfumagalli** [php-fig-rectified/fig-rectified-standards#20]
I don't understand what's the benefit we get by following this directive.

Actually, it makes code less clear and harder to "check at a glance", two reasons that made PSR discourage "Yoda conditions".

Taking your multi-line example, if I type:

```
if (
	($a == $b) &&
	($b == $c) ||
	(Foo::CONST == $d)
) {
	$a = $d;
}
```

* It looks prettier and properly aligned.
* It's easy to visually check if I have written the second comparison right. This is a very frequent coding situation. By putting the operator behind, I _immediately_:
  * see the parentheses are here,
  * see that I put "==" on both statements
  * see that I did not forget "$" prefixes.

Everything just aligns smoothly and a glance is enough to notice the most common, typical typos.

**Comment by dereuromark** [comment]
The most important part of multiple if conditions is the operator. That decides on the whole flow.
The comparators of the elements itself is not that important here.
Thus adding this at the beginning makes it way more easy to see this important piece of it.


```
if (
	($a == $b) 
	&& ($b == $c)
	|| (Foo::CONST == $d)
) {
	$a = $d;
}
```
*should* probably better read

```
if (
	($a == $b) && ($b == $c)
	|| (Foo::CONST == $d)
) {
	$a = $d;
}
```
Since the first two are linked together anyway. You can optionally also put another () around the first two for clarification.

**Comment by jrchamp** [comment]
Countering "It looks prettier and properly aligned.", variable names with one letter are typically a bad sign. In general, I should be able to read the code as a designed language sentence without needing to scroll to the top to figure out what $a and $b are. The word name which describes/self-documents the contents of the variable is unlikely to have a same number of letters and will therefore be unlikely to align when the variable naming is descriptive.

**Comment by dereuromark** [comment]
You misunderstand. My arguments are based on readability, exactly what you wanted to achieve. The looks prettier argument was in fact yours (Everything just aligns smoothly).
So please don't twist words. And alignment is irrelevant.

I already gave you the main reasons why it is smarter and more readable to do it the way described.
No need to discuss this now further.

**Comment by dereuromark** [comment]
As a SHOULD feel free to just ignore :)

**Comment by dfumagalli** [comment]
@dereuromark and @jrchamp  I have copied and pasted the very [PHP-FIG-rectified example](https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#multi-line-declarationconditionconcatenation), which should be authoritative when discussing about it.

The example shows single letter variables. In a realistic setting, the right aligment argument ceases to work, but the left and the == often keeps working. It's pretty common to find multiple conditions where the first variable name is repeated each line.

About the "_should_ probably better read", I quite agree on that. However I have checked some PSR documents and they show "aggressive statements splitting".
I'd rather code as per your example, which in my opinion helps avoiding unnecessary vertical walls of text. But is that PSR compliant?
